mibidata/tiff.py

```diff
@@ -282,32 +278,46 @@ def read(file, sims=True, sed=False, optical=False, label=False,
         _check_software(tif)
         for page in tif.pages:
             description, image_type = _page_description(page)
-            if sims and image_type == 'sims':
-                if _get_page_data(page, description, metadata, channels,
-                                  inc_channels):
-                    sims_data.append(page.asarray())
+            if sims and image_type == 'sims' and _include_page(
+                    description, masses, targets):
+                _get_page_data(page, description, metadata, channels)
+                sims_data.append(page.asarray())
             elif return_types.get(image_type):
                 to_return[image_type] = page.asarray()
     if sims:
-        if inc_channels and not sims_data:
+        if (targets or masses) and not sims_data:
```

Comment:
Does it make sense to restrict this to only masses *or* targets can be specified, in case someone specifies different lists for each? Also, what if someone passes a list of tuples of (mass, target)?

Reply:
I wondered about restricting to one or the other as well, and could be convinced to do that. I don't have a strong use-case for needing to specify a mix, the closest I got was a LOO-type experiment where you've extracted the empty channel when the target was left out, but in that case you could just specify masses across the board.

Reply:
I can't think of any use cases where specifying both would be needed, was more just concerned that the option should not be there to avoid potential issues when being used. For example, if someone accidentally specified two lists that were not the same, the data returned might not be what is expected and tough to track down why.

I do think it makes sense to be able to pass a list of tuples though, because I can image a scenario where someone uses a subset of `MibiImage.channels` to load a TIFF, in which case the list would be one of tuples.

Reply:
I updated to only allow masses _or_ targets to be specified.

I see you point about a list of tuples, but it's not hard to extra either channels or masses from that and I'm hesitant to litter the kwargs more than they already are, so can we put that off and possible revisit as part of the general channel formatting issue?